src/buttons.rs

```diff
@@ -0,0 +1,198 @@
+use atsamd_hal::clock::GenericClockController;
+use atsamd_hal::common::eic;
+use atsamd_hal::common::eic::pin::*;
+use atsamd_hal::gpio::*;
+use atsamd_hal::target_device::{interrupt, EIC, MCLK};
+
+use cortex_m::peripheral::NVIC;
+
+/// pushbuttons and joystick
+pub struct ButtonPins {
+    /// button1 pin
+    pub button1: Pc26<Input<Floating>>,
+    /// button2 pin
+    pub button2: Pc27<Input<Floating>>,
+    /// button3 pin
+    pub button3: Pc28<Input<Floating>>,
+
+    /// Joystick X
+    pub switch_x: Pd8<Input<Floating>>,
+    /// Joystick Y
+    pub switch_y: Pd9<Input<Floating>>,
+    /// Joystick Z
+    pub switch_z: Pd10<Input<Floating>>,
+    /// Joystick U
+    pub switch_u: Pd20<Input<Floating>>,
+    /// Joystick B
+    pub switch_b: Pd12<Input<Floating>>,
+}
+
+impl ButtonPins {
+    pub fn init(
+        self,
+        eic: EIC,
+        clocks: &mut GenericClockController,
+        mclk: &mut MCLK,
+        port: &mut Port,
+    ) -> ButtonController {
+        let clk = clocks.gclk1();
+        let mut eic = eic::init_with_ulp32k(mclk, clocks.eic(&clk).unwrap(), eic);
+        eic.button_debounce_pins(&[
+            self.button1.id(),
+            self.button2.id(),
+            self.button3.id(),
+            self.switch_x.id(),
+            self.switch_y.id(),
+            self.switch_z.id(),
+            self.switch_u.id(),
+            self.switch_b.id(),
+        ]);
+
+        // let mut b1 = self.button1.into_ei(port);
```

Comment:
Would you mind adding a quick comment stating why `b1` and friends are commented out? You told me on Matrix but I'm sure I'll forget.

Reply:
Done.